java/arcs/sdk/wasm/WasmHandle.kt

```diff
@@ -31,9 +31,50 @@ abstract class WasmHandleEvents<T>(
 ) : WasmHandle(name, particle) {
     protected val onUpdateActions: MutableList<(T) -> Unit> = mutableListOf()
 
+    /** Assign a function to be called when the handle is updated. */
     fun onUpdate(action: (T) -> Unit) {
         onUpdateActions.add(action)
     }
 
-    abstract fun getContent(): T
+    /**
+     * This allows combineUpdates to get the entit(ies) the handle points to by providing a
+     * consistent API. This is required as Singletons have fetch() while Collections have
+     * fetchAll().
+     */
+    protected abstract fun getContent(): T
+
+    // Note, this method is internal to the class so that getContent() can be protected,.
+    /**
+     * Internal combineUpdates method.
+     *
+     * @handle2 Handle that we want to track updates of
+     * @action Callback function
+     *
+     */
+    fun <U> combineUpdates(
```

Comment:
I think we talked quite explicitly that these utilities should live in something like a "Standard Library" on top of the core types. By doing this we achieve 2 things:
1) The core building blocks (handles) are super simple.
2) There is a rich set of operations you can perform with them.

Which is ideal.

Would you like to try to refactor it so that getContent() is an extension function private to the file in which combineUpdates() live? Is something like that possible?